### PR TITLE
Replicated event sourcing 

### PR DESCRIPTION
**Pull request by chbatey**
Merge of the feature branch back into master

It has just been rebased 

https://github.com/akka/akka/issues/29217

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/Member.scala [hunk 1]
```diff
@@ -41,9 +41,9 @@ class Member private[cluster] (
   }
   override def toString =
     if (dataCenter == ClusterSettings.DefaultDataCenter)
-      s"Member(address = $address, status = $status)"
+      s"Member(address = $address, status = $status, roles = $roles)"
```
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/Member.scala:44]: prettier
```
[${roles.mkString(", ")}]
```

File: akka-docs/src/main/paradox/project/examples.md [hunk 2]
```diff
@@ -61,11 +61,7 @@ from the events, or publish the events to other services.
 
 ## Multi-DC Persistence
 
-@java[@extref[Multi-DC Persistence example project](samples:akka-samples-persistence-dc-java)]
-@scala[@extref[Multi-DC Persistence example project](samples:akka-samples-persistence-dc-scala)]
-
-Illustrates how to use Lightbend's [Multi-DC Persistence](https://doc.akka.io/docs/akka-enhancements/current/persistence-dc/index.html)
-with active-active persistent entities across data centers.
+This commercial feature has now been superseeded by @ref[Replicated Event Sourcing](../typed/replicated-eventsourcing.md)
```
Comment (patriknw) [inline akka-docs/src/main/paradox/project/examples.md:64]: this can be updated again when the sample has been updated

**Review by patriknw — APPROVED** [review]
LGTM, I have reviewed this in previous PRs and dev branch. Great work @johanandren and @chbatey !

**Comment by patriknw** [comment]
Have we run this in a repeat jenkins job? Just to ensure that there are no surprises in multi-node tests.

**Comment by chbatey** [comment]
Not yet but i will before merging

File: akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ReplicatedShardingExtensionImpl.scala [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+          "Starting Replicated Event Sourcing sharding for replica [{}] (ShardType: [{}])",
+          replicaSettings.replicaId.id,
+          replicaSettings.entity.typeKey.name)
+        val regionOrProxy = sharding.init(replicaSettings.entity)
+        (
+          typeName,
+          replicaSettings.replicaId,
+          replicaSettings.entity.typeKey,
+          regionOrProxy,
+          replicaSettings.entity.dataCenter)
```
Comment (raboof) [inline akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ReplicatedShardingExtensionImpl.scala:60]: wonky formatting, but I guess that's scalafmt

**Review by raboof — COMMENTED** [review]
Some small docs comments, what I've seen looks good (but haven't looked at everything)

**Comment by chbatey** [comment]
Given that we don't plan to squash this branch i'll keep pushing feedback into a single commit after we have the other PR into this branch merged

**Comment by chbatey** [comment]
Running on https://jenkins.akka.io:8498/job/legacy-akka-multi-node-repeat4/

**Review by johanandren — APPROVED** [review]
Looked at the individual PRs done while I was vacationing and I think everything LGTM

**Comment by patriknw** [comment]
Let's fix the conflict and then merge this.

**Comment by patriknw** [comment]
I fixed the conflict. Let's have a few successful runs of multi-node tests before merging. Running at https://jenkins.akka.io:8498/job/legacy-akka-multi-node-repeat4/

**Comment by patriknw** [comment]
Several successful runs in the multi-node tests. Some failures in unrelated cluster.StressSpec https://github.com/akka/akka/issues/29512

**Comment by patriknw** [comment]
Merged 🎉 
Great work @johanandren and @chbatey !

**Pull request merged**